src/ansys/fluent/core/services/field_data.py

```diff
@@ -176,82 +253,20 @@ def get_surfaces_info(self) -> Dict[str, Dict]:
         }
         return info
 
-
-def unavailable_field_error_message(context: str, field_name: str) -> str:
-    """Error message for unavailable fields."""
-    return f"{field_name} is not a currently available {context}."
-
-
-class FieldNameError(ValueError):
-    """Exception class for errors in field name."""
-
-    pass
-
-
-class ScalarFieldNameError(FieldNameError):
-    """Exception class for errors in scalar field name."""
-
-    def __init__(self, field_name: str, allowed_values: List[str]):
-        """__init__ method of ScalarFieldNameError class."""
-        self.field_name = field_name
-        super().__init__(
-            allowed_name_error_message("scalar field", field_name, allowed_values)
-        )
-
-
-class VectorFieldNameError(FieldNameError):
-    """Exception class for errors in vector field name."""
-
-    def __init__(self, field_name: str, allowed_values: List[str]):
-        """__init__ method of VectorFieldNameError class."""
-        self.field_name = field_name
-        super().__init__(
-            allowed_name_error_message("vector field", field_name, allowed_values)
-        )
-
-
-class FieldUnavailable(RuntimeError):
-    """Exception class for when field is unavailable."""
-
-    pass
-
-
-class ScalarFieldUnavailable(FieldUnavailable):
-    """Exception class for when scalar field is unavailable."""
-
-    def __init__(self, field_name: str):
-        """__init__ method of ScalarFieldUnavailable class."""
-        self.field_name = field_name
-        super().__init__(unavailable_field_error_message("scalar field", field_name))
-
-
-class VectorFieldUnavailable(FieldUnavailable):
-    """Exception class for when vector field is unavailable."""
-
-    def __init__(self, field_name: str):
-        """__init__ method of VectorFieldUnavailable class."""
-        self.field_name = field_name
-        super().__init__(unavailable_field_error_message("vector field", field_name))
-
-
-class SurfaceNameError(ValueError):
-    """Exception class for errors in surface name."""
-
-    def __init__(self, surface_name: str, allowed_values: List[str]):
-        """__init__ method of SurfaceNameError class."""
-        self.surface_name = surface_name
-        super().__init__(
-            allowed_name_error_message("surface", surface_name, allowed_values)
+    def validate_fields(self, field_name: str):
+        allowed_fields = list(self.get_scalar_fields_info().keys()) + list(
```

Comment:
A `field_name` could be from either a scalar data request or a vector data request. In which situation we want to check whether the field is either an allowed scalar field or an allowed vector field?

Reply:
Our objective is to validate field name and field name is either scalar or vector. Therefore we have combined both an allowed scalar field and an allowed vector field names into single `allowed_fields`.
Then we are using general context of `field name` instead of specific scalar or vector field in error message.

Reply:
In which situation we want to check whether the field is either an allowed scalar field or an allowed vector field? - We are going to use this in Contour class in visualization repo.

![image](https://github.com/ansys/pyfluent/assets/106588300/d1baa4d4-d484-473a-a4f6-cc7410950d88)

Reply:
Will the above code raise exception when `self.field` is `"velocity "`, which is a vector field?

Reply:
@hpohekar here is your current `Contour` code
```
        allowed_fields = (
            solver.field_data.get_scalar_field_data.field_name.allowed_values()
        )
```

Reply:
> Will the above code raise exception when `self.field` is `"velocity"`, which is a vector field? - No.

Got it. We will use pre-defined error classes defined for FieldData using FieldInfo. 

We will remove current dependencies on FieldInfo.

Thank you.

Reply:
> @hpohekar here is your current `Contour` code
> 
> ```
>         allowed_fields = (
>             solver.field_data.get_scalar_field_data.field_name.allowed_values()
>         )
> ```

Right. I thought instead of creating dependencies on FieldData we can use FieldInfo methods directly in above cases, but as Mainak has mentioned it is not serving the purpose correctly.

We will update this now.

Thank you.

Reply:
Done. Thank you.